shadows/framework/src/main/java/org/robolectric/shadows/ShadowBitmap.java

```diff
@@ -899,4 +903,27 @@ void setBufferedImage(BufferedImage bufferedImage) {
   BufferedImage getBufferedImage() {
     return bufferedImage;
   }
+
+  void drawRect(RectF r, Paint paint) {
+    if (bufferedImage == null) {
+      return;
+    }
+    int[] pixels = ((DataBufferInt) bufferedImage.getRaster().getDataBuffer()).getData();
+
+    Graphics2D graphics2D = bufferedImage.createGraphics();
+    Rectangle2D r2d = new Rectangle2D.Float(r.left, r.top, r.right - r.left, r.bottom - r.top);
+    graphics2D.draw(r2d);
+    graphics2D.dispose();
+
+    double bottom=r2d.getY()+r2d.getHeight();
+    double right=r2d.getX()+r2d.getHeight();
+    if (r2d.getX() == 0 && r2d.getY() == 0 && right == r2d.getWidth()) {
+      Arrays.fill(pixels, 0, (int)(r2d.getWidth() * bottom), paint.getColor());
+      return;
+    }
+    for (double y = r2d.getY(); y < bottom; y++) {
+      Arrays.fill(
```

Comment:
I think above `Graphics2D` has changed pixels content and we don't need do it again.